### PR TITLE
Goon-style blob placement

### LINKED ISSUE #15296: blob help icon is broken
it doesn't work according to an ahelp
@ChangelingRain

### LINKED ISSUE #16056: Blobs can still burst on Toxins Test Satelite
I think it was MrPerson who stopped this from happening a long time ago and it seems to have broken again.

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscdel: Blob gamemode blobs will no longer burst from people.
rscadd: Instead, the blob gamemode spawns overminds, which can place their blob core underneath them in an unobserved area.
rscadd: The overminds can, until they have placed a blob core, move to any non-space, non-shuttle tile.
rscadd: If the overminds fail to place a blob core within a time limit, it will be automatically placed for them at a random blob spawnpoint.
tweak: The time before the overmind automatically places the core is slightly lower than the previous average time it'd take a blob to burst.
/:cl:

None of the code is taken from goon, by the way, I wrote all of it.
Fixes #15296 (by removing it)
Fixes #16056 (by preventing movement there)

### CODE REVIEW

**Comment by lzimann** [comment]
.>no more sabotage before as human

not a fan of it

**Comment by MMMiracles** [comment]
Being able to sabotage shit beforehand if you were fast enough was interesting but in the end it just always ended up being "release singulo/destroy RnD/space armory".

Also you should probably note that you decreased the amount of times blobs have before automatic placement in the changelog.

**Comment by ChangelingRain** [comment]
The average time is slightly lower, yeah, but I wasn't sure it was really notable, I guess I will.

**Comment by KorPhaeron** [comment]
Human sabotage was more or less the only variance between blob rounds.

**Comment by Incoming5643** [comment]
I misread the title and thought you were porting the blob smoothing goon has. Then I got disappointed.

I can't say I'm a fan of cutting out the shenanigans preburst phase either.

**Comment by AdamElTablawy** [comment]
>one blob places core in armory and eats all the guns

>the other one puts it in the AI sat/tcomms

>every round

**Comment by ChangelingRain** [comment]
Blobs can't place in the AI core because they can't reach it(and even if they could, there's an AI in there and it'd prevent placement)
Similarly, they can't place in the armory, because the warden can see them and it'll prevent placement.

**Comment by Shadowlight213** [comment]
New meta, hos and scientists stand watch over armory and rnd for the auto place time to deter blobs

**Comment by Zilenan91** [comment]
Perma blob will happen every round now

Though yeah it's probably a good thing that this happens. It felt like a cheap, hollow victory when a HoP blob burst with the entire armory and a subverted AI while I was a fellow blob. Nobody could fight us.

**Comment by Shadowlight213** [comment]
Subverting the AI as blob? Absolutely brilliant planning.

**Comment by YakumoChen** [comment]
no way to prevent the xray spam every round now, gg blob balance

**Comment by newfren** [comment]
If you want to prevent Xray spam then stop the minerals from getting back. Still not sure how I feel about this.

File: code/game/gamemodes/blob/overmind.dm [hunk 1]
```diff
@@ -18,8 +18,22 @@
...
+	var/placed = 0
+	var/base_point_rate = 2 //for blob core placement
+	var/manualplace_min_time = 600 //in deciseconds //a minute, to get bearings
+	var/autoplace_max_time = 3600 //six minutes, as long as should be needed
+
+/mob/camera/blob/New(loc, placed = 0, mode_made = 0)
+	if(placed) //we already have a core!
+		manualplace_min_time = 0
+		autoplace_max_time = 0
+		placed = 1
```
Comment (phil235) [inline code/game/gamemodes/blob/overmind.dm]: if(placed) placed = 1 ?

Comment (ChangelingRain) [inline code/game/gamemodes/blob/overmind.dm]: whoop yeah I should fix that

File: code/game/gamemodes/blob/blobs/core.dm [hunk 2]
```diff
@@ -15,16 +15,16 @@
 	var/point_rate = 2
 
 
-/obj/effect/blob/core/New(loc, var/h = 200, var/client/new_overmind = null, var/new_rate = 2)
+/obj/effect/blob/core/New(loc, var/client/new_overmind = null, var/new_rate = 2, var/placed = 0)
```
Comment (phil235) [inline code/game/gamemodes/blob/blobs/core.dm]: don't need "var/" for the arguments.

**Comment by Iamgoofball** [comment]
yeah you can stop xray spam if you pop in the mining dock/nearby R&D

**Comment by PKPenguin321** [comment]
do we really want blobs popping in mining/R&D every round tho
also does the blob overmind get night vision? it should so that it can still pop in maint

**Comment by ChangelingRain** [comment]
The meta is already 'disabling R&D and or mining' so this isn't that big of a change on that side.
And yeah it already has nightvision.

**Comment by PKPenguin321** [comment]
disabling them != popping in them

**Comment by AdamElTablawy** [comment]
>blob pops near mining
>minerals never come
>gg

>blob pops in tcomms
>other guy distractionpops in R&D

>gg

**Comment by lzimann** [comment]
:-1:

**Comment by Aloraydrel** [comment]
Rip blob being nerfed every pr

**Comment by Incoming5643** [comment]
Why

**Pull request merged**